### ISSUE TITLE
(ES6) warn of "const" used inside a for loop

### ISSUE THREAD

**Issue by gre** [eslint/eslint#3608]
I can't find a reference for this but it seems that using a `const` inside a `for` block is wrong (in Babel, it seems to always have the same value once first evaluated).

eslint should warn/error about this.

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by gre** [comment]
```
for (var i=0; i<9; i++) {
  const j = i;
  console.log(j);
}
```

if you try to run this in Chrome Console, it display 9 times, "0",
if you transpile with Babel it displays values from 0 to 9.

Not sure who is right.

**Comment by michaelficarra** [comment]
Babel is right.

**Comment by nzakas** [comment]
Per @michaelficarra's comment, there's nothing wrong with using `const` inside a loop, so closing
